src/dataset/mod.rs

```diff
@@ -198,6 +198,14 @@ pub trait Records: Sized {
     fn nfeatures(&self) -> usize;
 }
 
+/// MultiTask trait
+pub trait MultiTaskTarget: Sized {
+    type Elem;
+
+    fn nsamples(&self) -> usize;
+    fn ntasks(&self) -> usize;
+}
+
```

Comment:
Why do you need to have this trait? Can't you just call `nrows()` and `ncols()` on 2D arrays directly?

Reply:
I removed it.